File: cli-development/templates/go-cobra/cmd/create.go
package cmd

import (
	"fmt"
	"time"

	"github.com/fatih/color"
	"github.com/spf13/cobra"
)

var (
	template  string
	skipInstall bool
)

var createCmd = &cobra.Command{
	Use:   "create [name]",
	Short: "Create a new project",
	Long: `Create a new project from a template.

Example:
  mycli create myapp
  mycli create myapp --template react`,
	Args: cobra.ExactArgs(1),
	Run: func(cmd *cobra.Command, args []string) {
		name := args[0]
		createProject(name)
	},
}

func init() {
	createCmd.Flags().StringVarP(&template, "template", "t", "default", "Template to use")
	createCmd.Flags().BoolVar(&skipInstall, "skip-install", false, "Skip npm install")
}

func createProject(name string) {
	cyan := color.New(color.FgCyan).SprintFunc()
	green := color.New(color.FgGreen).SprintFunc()

	fmt.Printf("\n📦 Creating project: %s\n\n", cyan(name))
	fmt.Printf("  Template: %s\n", green(template))

	// プロジェクト作成シミュレーション
	steps := []string{
		"Creating directory structure",
		"Copying template files",
		"Configuring project",
	}

	if !skipInstall {
		steps = append(steps, "Installing dependencies")
	}

	for _, step := range steps {
		fmt.Printf("  %s...\n", step)
		time.Sleep(500 * time.Millisecond)
	}

	fmt.Printf("\n%s Project created successfully!\n\n", green("✓"))
	fmt.Printf("%s\n", cyan("Next steps:"))
	fmt.Printf("  cd %s\n", name)
	if skipInstall {
		fmt.Println("  npm install")
	}
	fmt.Println("  npm run dev\n")
}

File: _legacy/automation/cli-development/templates/python-typer/src/cli/commands/create.py
"""Create command."""
import typer
from rich.console import Console
from rich.progress import track
from rich.panel import Panel
import time

app = typer.Typer()
console = Console()

@app.command()
def project(
    name: str = typer.Argument(..., help="Project name"),
    template: str = typer.Option("default", "--template", "-t", help="Template to use"),
    typescript: bool = typer.Option(True, "--typescript/--no-typescript"),
    install: bool = typer.Option(True, "--install/--no-install", help="Install dependencies"),
) -> None:
    """
    Create a new project from a template.

    Example:
        mycli create myapp
        mycli create myapp --template react --no-typescript
    """
    console.print(Panel.fit(
        f"[cyan]Creating project: {name}[/cyan]",
        title="Project Generator",
        border_style="cyan"
    ))

    console.print(f"Template: [green]{template}[/green]")
    console.print(f"TypeScript: [green]{typescript}[/green]")

    # プロジェクト作成処理
    tasks = [
        "Creating directory structure",
        "Copying template files",
        "Configuring project",
    ]

    if install:
        tasks.append("Installing dependencies")

    for task in track(tasks, description="Setting up project"):
        time.sleep(0.5)

    console.print("\n[bold green]✓ Project created successfully![/bold green]\n")
    console.print("[cyan]Next steps:[/cyan]")
    console.print(f"  cd {name}")
    if not install:
        console.print("  npm install")
    console.print("  npm run dev")
